pkgs/by-name/sa/satty/package.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, wrapGAppsHook4
+, cairo
+, gdk-pixbuf
+, glib
+, gtk4
+, libadwaita
+, pango
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "satty";
+  version = "main-2023-11-21";
```

Comment:
```suggestion
  version = "unstable-2023-11-21";
```

Reply:
Why are you packaging the unstable version at all? There are tagged releases.

Reply:
The tagged release would mean vendoring the lock file and carrying a patch. The fix for that is only after the tagged release.

Reply:
This should work just fine.
```nix
{
  version = "0.7.0";

  src = fetchFromGitHub {
    owner = "gabm";
    repo = "Satty";
    rev = "v${version}";
    hash = "sha256-x2ljheG7ZqaeiPersC/e8Er2jvk5TJs65Y3N1GjTiNU=";
  };

  cargoPatches = [
    (fetchpatch {
      name = "fix-Cargo.lock";
      url = "https://github.com/gabm/Satty/commit/39be6ddce264552df971e949a6a3175b102530b2.patch";
      hash = "sha256-GUHupZE1A7AmXvZ8WvRzBkQyH7qlMTetBjHuakfIZ7w=";
    })
  ];

  cargoHash = "sha256-0GsbWd/gpKZm7nNXkuJhB02YKUj3XCrSfpRA9KBXydU=";
}
```

Reply:
Ah, thanks! Yes that does seem like it would work.